core/blockchain.go

```diff
@@ -1539,10 +1547,11 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 					bc.triegc.Push(root, number)
 					break
 				}
-				triedb.Dereference(root.(common.Hash))
+				go triedb.Dereference(root.(common.Hash))
 			}
 		}
 	}
+	wg.Wait()
```

Comment:
why wait here instead of line 1487

Reply:
I want to `state.Commit` and `blockBatch.Write()` run concurrently